### ISSUE TITLE
Can TryGetValue Return None for Null Value Instead of Throwing an Exception?

### ISSUE THREAD

**Issue by pkirch-swdev** [louthy/language-ext#901]
Currently the IDictionary.TryGetValue(key) extension method throws an exception when the accessed item value is null. Can this be changed to return None instead of throwing an exception?

**Comment by pkirch-swdev** [comment]
public static Option<V> TryGetValue<K, V>(this IDictionary<K, V> self, K Key)
        {
            V value;
            return self.TryGetValue(Key, out value)
                ? Some(value)  // <- make this Optional(value) instead?
                : None;
        }

**Comment by louthy** [comment]
Yes, I think that’s probably reasonable.  Feel free to submit a PR 👍
